src/main/resources/test_files/runtime-spark/RumbleML/AnnotateFunction/MLAnnotateForEstimator1.jq

```diff
@@ -7,7 +7,7 @@ let $local-data := (
     {"id": 4, "col1": 9.1, "col2": 9.1, "col3": 9.1},
     {"id": 5, "col1": 9.2, "col2": 9.2, "col3": 9.2}
 )
-let $df-data := annotate($local-data, {"id": "integer", "col1": "double", "col2": "double", "col3": "double"})
+let $df-data := annotate($local-data, {"id": "integer", "col1": "decimal", "col2": "decimal", "col3": "decimal"})
 
 let $est := get-estimator("KMeans")
```

Comment:
@ghislainfourny, following our discussion today, the reason decimal types here work with KMeans here is the underlying VectorAssembler usage. At line15, these decimal columns are vectorized to form a featuresVector. Having a sparse vector of doubles, KMeans can operate on them. The debugging screenshot is given below:

<img width="840" alt="image" src="https://user-images.githubusercontent.com/13052327/76415018-b7ea8400-6398-11ea-830f-7fae9389a869.png">

To summarize,  I'm assuming most of the ML components do not support operations on decimalType. VectorAssembler however, is an exceptional transformer that is capable of operating on Decimals. We will note if we come across any more Decimal supporting transformers or estimators along the way.

In the current state, if a decimal supported exception is thrown upon calling a fit or a transform, this will be caught with regex and will throw an appropriate exception.

Reply:
Perfect. Thank you for the clarification!